.github/workflows/release-code-freeze.yml

```diff
@@ -94,7 +94,7 @@ jobs:
           SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
         with:
           # For posting a rich message using Block Kit (https://api.slack.com/messaging/interactivity)
-          payload: |       
+          payload: |
```

Comment:
Can we slightly update the text from the Slack ping? I personally wrote it but I think we can make it more concise and clear, and we can also mention that we now already built the zip file + run the smoke tests.

Here is what I had in mind: [Slack Block Kit Builder preview](https://app.slack.com/block-kit-builder/T024FN1V2#%7B%22blocks%22:%5B%7B%22type%22:%22header%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Code%20freeze%20notification%22,%22emoji%22:true%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22The%20automated%20workflow%20just%20did%20the%20following%20actions:%22%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22%E2%80%A2%20Create%20a%20release%20branch%20%60release/x.y.z%60%20%5Cn%20%E2%80%A2%20Raise%20a%20Pull%20Request%20using%20%60trunk%60%20as%20the%20base%20%5Cn%20%E2%80%A2%20Build%20a%20zip%20file%20and%20run%20the%20E2E%20tests%20against%20it%22%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22If%20you%20absolutely%20need%20to%20include%20a%20PR%20that%20was%20not%20merged%20in%20%60develop%60%20yet,%20please%20consult%20the%20release%20lead.%22%7D%7D,%7B%22type%22:%22section%22,%22fields%22:%5B%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Version:*%5Cn$%7B%7B%20env.RELEASE_VERSION%20%7D%7D%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Planned%20date:*%5Cn%3Chttps://wcpay.wordpress.com/dev-resources/wc-payments-release-calendar/%7C$%7B%7B%20env.RELEASE_DATE%20%7D%7D%3E%22%7D%5D%7D%5D%7D)

Note: replace `release/x.y.z` by the corresponding env variables in the code, and remove `GITHUB_TAG_URL` env from the job, since it will not used anymore

Basically, keep it short with list items to enumerate what the workflow has done, and remove the link behind the `version` (the GH release doesn't exist yet when the ping is sent, so the link returns a 404).

Nice to have: 

- add a new section with 2 fields `Release zip file` and `Smoke tests run`, below the `Version` and `Planned date` section

OR (if not possible; not clear if we can retrieve a working link for the artifact before the workflow finishes)

- add a PR comment to the `release/x.y.z>trunk` PR with a link to the zip file and a link to the smoke tests run

I think it's best if we ship what you already built, so do you mind creating a new issue for this "nice to have" improvement?

Reply:
> add a new section with 2 fields Release zip file and Smoke tests run, below the Version and Planned date section

@jbordonado We can link to the GH action run, which would also help with downloading the artifact & see the status of smoke test. It won't be possible to get the artifact data until the workflow is finished. 

I tried fetching the artifact data within the workflow and I got `{"total_count":0,"artifacts":[]}`

But after the workflow is finished I got:

```
{
  "total_count": 1,
  "artifacts": [
    {
      "id": 592122679,
      "node_id": "MDg6QXJ0aWZhY3Q1OTIxMjI2Nzk=",
      "name": "woocommerce-payments",
      "size_in_bytes": 14341765,
      "url": "https://api.github.com/repos/Automattic/woocommerce-payments/actions/artifacts/592122679",
      "archive_download_url": "https://api.github.com/repos/Automattic/woocommerce-payments/actions/artifacts/592122679/zip",
      "expired": false,
      "created_at": "2023-03-10T04:26:27Z",
      "updated_at": "2023-03-10T04:26:28Z",
      "expires_at": "2023-03-24T04:25:39Z",
      "workflow_run": {
        "id": 4381298159,
        "repository_id": 173384525,
        "head_repository_id": 173384525,
        "head_branch": "devops/automate-smoke-tests",
        "head_sha": "bc3dd1955540d17d339d12d086146fa95194b42e"
      }
    }
  ]
}
```

Here is the GitHub documentation related to artifacts.
> During a workflow run, you can use the [download-artifact](https://github.com/actions/download-artifact) action to download artifacts that were previously uploaded in the same workflow run. After a workflow run has been completed, you can download or delete artifacts on GitHub or using the REST API.

Reply:
So the options we have are:

1. Add link to the workflow run in the slack notification which can be used to download the artifact & check smoke test status.
2. Move slack notification to a new workflow which can be triggered when required and runs after the current workflow is finished. Then we can fetch the artifact data and send it with the slack ping. I don't wish to complicate things more, so this is my least preferred option atm.

Reply:
I will create new issues to improve the slack notification. We can look at them later

Reply:
Option 1 is fine. But as I said, all of these can happen later in a new set of issues, I just wanted to lay out my ideas when I started to review the PR.

If we can just adjust the wording like on the link I shared, that would be already very good